common/djangoapps/enrollment/views.py

```diff
@@ -63,7 +101,38 @@ def get(self, request, course_id=None, user=None):
 
 
 class EnrollmentCourseDetailView(APIView):
-    """ Enrollment API View for viewing course enrollment details. """
+    """ 
+        **Use Cases**
+            
+            Get enrollment details for a course.
+
+        **Example Requests**:
+
+            GET /api/enrollment/v1/course/{course_id}
+
+
+        **Response Values**
+
+            A collection of course enrollments for the user, or for the newly created enrollment. Each course enrollment contains:
+
+                * course_id: The unique identifier of the course.
+
+                * enrollment_end: The date and time after which users cannot enroll for the course.
+
+                * course_modes: An array of data about the enrollment modes supported for the course. Each enrollment mode collection includes:
+
+                        * slug: The short name for the enrollment mode.
+                        * name: The full name of the enrollment mode.
+                        * min_price: The minimum price for which a user can enroll in this mode.
+                        * suggested_prices: A list of suggested prices for this enrollment mode.
+                        * currency: The currency of the listed prices.
+                        * expiration_datetime: The date and time after which users cannot enroll in the course in this mode.
+                        * description: A description of this mode.
+
+                * enrollment_start: The date and time that users can begin enrolling for the course.
+
+                * invite_only: Whether students must be invited to enroll in the course; true or false.
+    """
 
     authentication_classes = []
```

Comment:
Note that this view does not have authentication classes, and it will be available to anyone. (With the intent to use for About Pages)